scripts/cafe-process.py

```diff
@@ -11,28 +11,67 @@
 from argparse import ArgumentParser
 
 from twisted.internet import task
-from twisted.internet.defer import gatherResults, inlineCallbacks
+from twisted.internet.defer import (
+    DeferredSemaphore, gatherResults, inlineCallbacks)
 from twisted.internet.utils import getProcessOutputAndValue
 
 
+def get_root():
+    from test_repo import autoscale
+    return autoscale.__path__[0]
+
+
+def get_cafe_args(packages, modules, excludes):
+    if modules:
+        proc_args = []
+        if packages:
+            proc_args.extend(['-p'] + packages)
+        return [proc_args + ['-m', module]
+                for module in modules if module not in excludes]
```

Comment:
I think this and the next section assume that excludes is not `None`, which it would be if excludes were not provided, and hence there would be an exception raised.  Should we set a default, like `excludes = () if excludes is None else excludes`?